### DISCUSSION TITLE
Seperating out Models and Naming Convention

### DISCUSSION THREAD

**Discussion by microtechno9000** [automatic-ripping-machine/automatic-ripping-machine#647] [Q&A]
I have started separating out some of the models from the models.py file into separate files to make it easier to view/maintain some of the models.

Ran into an issue, in separating out the SystemInfo model into a SystemInfo.py file in the routes.py.

changes
```
from models import SystemInfo, SystemDrives
...
server = SystemInfo.query.filter_by(id="1").first()
```
Doesn't work, with an error
`routes.handle_exception Error: module 'models.SystemInfo' has no attribute 'query'`

changing the line to the below does work
`server = SystemInfo.SystemInfo.query.filter_by(id="1").first()`

If we are going to break out each of the classes/models into separate files, how do we go about naming them, or calling them? Seems a bit redundant to call 'filename.classname' everytime we need to call a module.

Thoughts? or is there some python guidance/styling that I haven't found?

**Comment by shitwolfymakes** [comment]
I believe the fix for that is adding `from . import SystemInfo` to the `__init__.py` file in the models directory.

Longstanding best practice is software engineering is **1 class per file**, no matter how small. This is important because classes have a habit of growing over time.

However, it would also behoove us to enforce a policy of **1 route per file**, for the same reason.

**Reply by microtechno9000** [reply]
I will give that a try for the models to start breaking them out.

For the routes, I have raised a small  PR #650, breaking out some of the routes into blueprints
